### PR TITLE
ui things

### PR DESCRIPTION
**Pull request by clarisaleu**
sound effects

### CODE REVIEW

**Review by jocelynshen — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -62,7 +67,8 @@
             android:screenOrientation="portrait"/>
         <activity
             android:name=".HomeActivity"
-            android:configChanges="keyboard|keyboardHidden" />
+            android:configChanges="orientation|keyboard|keyboardHidden"
```
Comment (ajankenm) [inline app/src/main/AndroidManifest.xml:70]: According to https://developer.android.com/guide/topics/manifest/activity-element#config "configChanges" are for "Lists configuration changes that the activity will handle itself", are you handling orientation changes or keyboard changes yourself?

File: app/src/main/java/com/example/footprnt/Discover/DiscoverFragment.java [hunk 2]
```diff
@@ -34,7 +34,8 @@
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
+        getActivity().setRequestedOrientation(
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Discover/DiscoverFragment.java:37]: Is this necessary if you only allow portrait in the manifest?

File: app/src/main/java/com/example/footprnt/HomeActivity.java [hunk 3]
```diff
@@ -30,10 +33,13 @@
 public class HomeActivity extends AppCompatActivity {
 
     final FragmentManager mFragmentManager = getSupportFragmentManager();
-    final Fragment mFragment1 = new MapFragment();
-    final Fragment mFragment2 = new DiscoverFragment();
-    final Fragment mFragment3 = new ProfileFragment();
-
+BottomNavigationView navView;
+ View shaddow;
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/HomeActivity.java:37]: mShadow

File: app/src/main/java/com/example/footprnt/HomeActivity.java [hunk 4]
```diff
@@ -30,10 +33,13 @@
...
+    Fragment mFragment1;
+    Fragment mFragment2;
+    Fragment mFragment3;
+    ViewPager viewPager;
+    MenuItem prevMenuItem;
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/HomeActivity.java:42]: mPrevMenuItem

File: app/src/main/java/com/example/footprnt/HomeActivity.java [hunk 5]
```diff
@@ -42,37 +48,90 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (resultCode == Constants.RELOAD_USERPROFILE_FRAGMENT_REQUEST_CODE) {
             mFragment3.onActivityResult(requestCode, resultCode, data);
         }
+        if(resultCode == 302){
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/HomeActivity.java:51]: Make 302 a constant

File: app/src/main/java/com/example/footprnt/HomeActivity.java [hunk 6]
```diff
@@ -42,37 +48,90 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
...
+                        viewPager.setCurrentItem(0);
                         break;
+
                 }
-                mFragmentManager.beginTransaction().replace(R.id.flContainer, fragment).commit();
                 return true;
             }
         });
         navView.setSelectedItemId(R.id.navigation_home);
+
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/HomeActivity.java:89]: Why 2 empty lines?

File: app/src/main/java/com/example/footprnt/HomeActivity.java [hunk 7]
```diff
@@ -42,37 +48,90 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
...
+
+        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
+            @Override
+            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/HomeActivity.java:94]: You should never leave a method empty. Add a "// TODO" to indicate what work will be done or add a "// Empty method" to indicate that it's empty on purpose

File: app/src/main/java/com/example/footprnt/Profile/Adapters/MultiViewAdapter.java [hunk 8]
```diff
@@ -96,12 +96,13 @@ public int getItemViewType(int position) {
             return USER_INFO;
         } else if (items.get(position) instanceof ArrayList) {
             return STAT;
-        }  else if(items.get(position) instanceof String){
+        } else if (items.get(position) instanceof String) {
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Profile/Adapters/MultiViewAdapter.java:99]: Whoo!!!!

File: app/src/main/java/com/example/footprnt/Profile/Adapters/MultiViewAdapter.java [hunk 9]
```diff
@@ -96,12 +96,13 @@ public int getItemViewType(int position) {
...
             return NO_POSTS;
         }
         return -1;
     }
 
+    LayoutInflater inflater;
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Profile/Adapters/MultiViewAdapter.java:105]: Move this to the top of the class and name it mInflater.

File: app/src/main/java/com/example/footprnt/Profile/Adapters/MultiViewAdapter.java [hunk 10]
```diff
@@ -174,10 +175,21 @@ private void configurePostViewHolder(final PostViewHolder vh1, final int positio
         final Post post = (Post) items.get(position);
         if (post != null) {
             vh1.getRootView().setTag(post);
+            StringBuilder sb = new StringBuilder();
             String cityName = post.getCity();
+            if (cityName != null) {
+                sb.append(cityName + ", ");
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Profile/Adapters/MultiViewAdapter.java:181]: Perfect world would have
  sb.append(cityName).append(", ");

File: app/src/main/java/com/example/footprnt/Profile/Adapters/MultiViewAdapter.java [hunk 11]
```diff
@@ -258,25 +263,51 @@ private void configureStatViewHolder(final StatViewHolder vh3, final int positio
         final HashMap<String, Integer> countries = stats.get(1);
         final HashMap<String, Integer> continents = stats.get(2);
 
-        if (cities != null) {
+        if (cities != null && cities.size() < ProfileConstants.totalNumCities) {
             setUpPieChart(vh3.getPieChartCity(), cities.size(), ProfileConstants.totalNumCities, "Visited Cities");
+        } else {
+
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Profile/Adapters/MultiViewAdapter.java:269]: Why is there an empty line here?

File: app/src/main/java/com/example/footprnt/Profile/Adapters/MultiViewAdapter.java [hunk 12]
```diff
@@ -258,25 +263,51 @@ private void configureStatViewHolder(final StatViewHolder vh3, final int positio
...
-        if (countries != null) {
-            setUpPieChart(vh3.getPieChartCountry(), countries.size(), ProfileConstants.totalNumContinents, "Visited Countries");
-        }
 
-        if (continents != null) {
-            setUpPieChart(vh3.getPieChartContinent(), continents.size(), ProfileConstants.totalNumCountries, "Visited Continents");
+        if (countries != null && countries.size() < ProfileConstants.totalNumCountries) {
+            setUpPieChart(vh3.getPieChartCountry(), countries.size(), ProfileConstants.totalNumCountries, "Visited Countries");
+        } else {
+            View view = vh3.getRootView().findViewById(R.id.pieChartCountry);
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Profile/Adapters/MultiViewAdapter.java:284]: If view could be null, we should indicate by marking this @Nullable

File: app/src/main/java/com/example/footprnt/Profile/EditPost.java [hunk 13]
```diff
@@ -96,6 +107,9 @@ public void done(final ParseObject object, ParseException e) {
                                 @Override
                                 public void done(ParseException e) {
                                     object.deleteInBackground();
+                                    Intent it = new Intent();
+                                    it.putExtra("position", position);
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Profile/EditPost.java:111]: "position" should be a constant

File: app/src/main/java/com/example/footprnt/Profile/EditPost.java [hunk 14]
```diff
@@ -50,36 +51,46 @@ protected void onCreate(Bundle savedInstanceState) {
...
-            Glide.with(this).load(R.drawable.ic_add_photo).into(mIvPicture);
         }
 
         // Set view window to be smaller for pop up effect
         DisplayMetrics displayMetrics = new DisplayMetrics();
         getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
         int width = displayMetrics.widthPixels;
         int height = displayMetrics.heightPixels;
-        getWindow().setLayout((int) (width * .8), (int) (height * .7));
+        getWindow().setLayout((int) (width * .8), (int) (height * .8));
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Profile/EditPost.java:93]: Digging through this class to find the .7 to change to a .8 is the reason why I push you to make these class constants for easier configuration as you test or change your mind.

File: app/src/main/java/com/example/footprnt/ViewPagerAdapter.java [hunk 15]
```diff
@@ -0,0 +1,30 @@
+package com.example.footprnt;
+
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentPagerAdapter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ViewPagerAdapter extends FragmentPagerAdapter {
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/ViewPagerAdapter.java:10]: Javadoc please :)

**Pull request merged**